Get rid of some uses of "execute_script" in feature specs (#4334)

ragesoss opened the pull request:
I made some progress here with #4334

This set of changes replaces all but four of the instances of `execute_script` in the Capybara tests, using the Capybara DSL instead. Some of these changes might not have worked with earlier Capybara version/driver combinations, but they should work with the current apparition driver.

Two of the remaining instances should perhaps remain in place. One is disabling of jQuery effects, which was done (11 years ago) just to make the tests more reliable, and might not even be necessary anymore, but probably can't be replaced if it is necessary. The other is for providing browser location. I haven't looked into whether this can be done instead with the Capybara DSL. If I find a way, I'll add another commit.

For two others, I was able to run the tests locally after removing or replacing `execute_script` usage, but I had to revert them because they still cause failures on CI.

ragesoss committed:
Refactor "I scroll" web step to avoid using execute_script

The apparition driver does not support "scroll_by" without going through "execute_script", but we can better express the intention of this test and make the scrolling action a bit more general by replacing "scroll a bit" with "scroll to [some element]" — in this case, we need to scroll until the "Change" button is visible so that we can press it.

This change_settings.feature spec is the only place that used "I scroll a bit".

ragesoss committed:
Remove unused custom web step

It looks like this web step is no longer used anywhere. It was implemented for the "Feature: inifinite scroll" test, but that no longer exists.

ragesoss committed:
Remove unnecessary `execute_script` call in feature spec

This call to `execute_script` would directly call jQuery's `sortable` function on an element to make it sortable. However, this isn't necessary; the test can still drag the "Cat People" aspect to the desired position without it.

ragesoss committed:
Replace "execute_script" for triggering hovercards

In addition to getting rid of an "execute_script" instance, this approach is slightly higher fidelity, as it renders the hovercard at the correct place on the page and doesn't reach into jQuery to trigger the hovercard.

ragesoss committed:
Replace "execute_script" for filling in Close Account modal

Using "fill_in" seems to work fine with the apparition driver, and will hopefully also work on CI now.

ragesoss committed:
Replace "execute_script" in spec scrolling down on notifications

ragesoss committed:
Replace "execute_script" in publisher_steps definitions

ragesoss committed:
Replace "evaluate_script" for toggling a Bootstrap switch in spec

ragesoss committed:
Replace "execute_script" for uploading images in specs

diaspora-code-review commented on features/step_definitions/hovercard_steps.rb:
```diff
@@ -14,7 +14,7 @@
 
 When(/^I deactivate the first hovercard$/) do
-  page.execute_script("$('.hovercardable').first().trigger('mouseleave');")
+  find("input#q").click # Click something else instead — e.g., search — to deactive it
```
Style/AsciiComments: Use only ascii symbols in comments.

ragesoss committed:
Fix comment style violation

ragesoss commented:
Alas, it looks like the hovercard cuke fails on CI with the click version. It passes for me locally in both headless and headed modes. I'll revert that one and see if the rest are okay.

ragesoss committed:
Revert "Replace "execute_script" for triggering hovercards"

This reverts commit e7dc4eca9ecdb025c01b02890e8e075ebc1af9cc.

ragesoss commented:
Hmm... 1 failure is now from `cucumber features/desktop/hovercards.feature:63 # Scenario: Hovercards contain sharing status when not sharing`, and three are from `cucumber features/mobile/closes_account.feature:7 # Scenario: user closes account`.

I'll check whether any of the changes in this PR affect those scenarios.

ragesoss committed:
Undo change to aspects_steps.rb that breaks the build

When I removed this and tested it locally, the feature spec still passed, but it is failing on CI, so it looks like we still need this `execute_script` use until we find a better Capybara strategy.

ragesoss committed:
Undo another Capybara refactor that causes CI to break

Alas.

SuperTux88 reviewed (approved):
Thanks for having a look at this and again sorry it took so long. And again thanks for the great commit message :+1: 

Sad some of the changes still failed in CI but worked locally, but maybe we can find a solution for those in the future.

SuperTux88 commented:
Merged as ... many commits to `next-minor`, thanks again :cookie:
